### ISSUE TITLE
Feature request: optional dependencies

### ISSUE THREAD

**Issue by DarwinAwardWinner** [doublep/eldev#31]
One nice-to-have feature would be a way to explicitly declare an optional dependency, which Eldev will try to install but then keep going regardless of whether the install succeeds or fails. And then you could also have a command-line option to disable all optional dependencies (e.g. in order to test that the package's base functionality doesn't have any unintentional hard dependencies on optional packages). But like I said, this is just nice-to-have. Most packages developers probably don't need anything like this.

So instead of:

```elisp
(when (version<= "25.1" emacs-version)
  (eldev-add-extra-dependencies 'test 'helm))
```

I could say:

```elisp
(eldev-add-extra-optional-dependencies 'test 'helm)
```

And then when testing on Emacs 24, Eldev would see that helm can't be installed and just skip it and keep going.

_Originally posted by @DarwinAwardWinner in https://github.com/doublep/eldev/issues/29#issuecomment-766399496_

**Comment by doublep** [comment]
Released in 0.9. See [documentation](https://github.com/doublep/eldev#optional-additional-dependencies) (also the [previous section](https://github.com/doublep/eldev#extended-dependency-format)).

**Comment by DarwinAwardWinner** [comment]
Excellent! Thanks very much for implementing this.
